sympy/solvers/diophantine/tests/test_diophantine.py

```diff
@@ -52,150 +85,204 @@ def test_nosols():
+        [x, y, z],
+        {x ** 2: 1, y ** 2: 1, z ** 2: 1},
+        "homogeneous_ternary_quadratic_normal",
+    )
 
 
 def test_linear():
     assert diop_solve(x) == (0,)
-    assert diop_solve(1*x) == (0,)
-    assert diop_solve(3*x) == (0,)
+    assert diop_solve(1 * x) == (0,)
+    assert diop_solve(3 * x) == (0,)
     assert diop_solve(x + 1) == (-1,)
-    assert diop_solve(2*x + 1) == (None,)
-    assert diop_solve(2*x + 4) == (-2,)
+    assert diop_solve(2 * x + 1) == (None,)
+    assert diop_solve(2 * x + 4) == (-2,)
     assert diop_solve(y + x) == (t_0, -t_0)
     assert diop_solve(y + x + 0) == (t_0, -t_0)
     assert diop_solve(y + x - 0) == (t_0, -t_0)
-    assert diop_solve(0*x - y - 5) == (-5,)
-    assert diop_solve(3*y + 2*x - 5) == (3*t_0 - 5, -2*t_0 + 5)
-    assert diop_solve(2*x - 3*y - 5) == (3*t_0 - 5, 2*t_0 - 5)
-    assert diop_solve(-2*x - 3*y - 5) == (3*t_0 + 5, -2*t_0 - 5)
-    assert diop_solve(7*x + 5*y) == (5*t_0, -7*t_0)
-    assert diop_solve(2*x + 4*y) == (2*t_0, -t_0)
-    assert diop_solve(4*x + 6*y - 4) == (3*t_0 - 2, -2*t_0 + 2)
-    assert diop_solve(4*x + 6*y - 3) == (None, None)
-    assert diop_solve(0*x + 3*y - 4*z + 5) == (4*t_0 + 5, 3*t_0 + 5)
-    assert diop_solve(4*x + 3*y - 4*z + 5) == (t_0, 8*t_0 + 4*t_1 + 5, 7*t_0 + 3*t_1 + 5)
-    assert diop_solve(4*x + 3*y - 4*z + 5, None) == (0, 5, 5)
-    assert diop_solve(4*x + 2*y + 8*z - 5) == (None, None, None)
-    assert diop_solve(5*x + 7*y - 2*z - 6) == (t_0, -3*t_0 + 2*t_1 + 6, -8*t_0 + 7*t_1 + 18)
-    assert diop_solve(3*x - 6*y + 12*z - 9) == (2*t_0 + 3, t_0 + 2*t_1, t_1)
-    assert diop_solve(6*w + 9*x + 20*y - z) == (t_0, t_1, t_1 + t_2, 6*t_0 + 29*t_1 + 20*t_2)
+    assert diop_solve(0 * x - y - 5) == (-5,)
+    assert diop_solve(3 * y + 2 * x - 5) == (3 * t_0 - 5, -2 * t_0 + 5)
+    assert diop_solve(2 * x - 3 * y - 5) == (3 * t_0 - 5, 2 * t_0 - 5)
+    assert diop_solve(-2 * x - 3 * y - 5) == (3 * t_0 + 5, -2 * t_0 - 5)
+    assert diop_solve(7 * x + 5 * y) == (5 * t_0, -7 * t_0)
+    assert diop_solve(2 * x + 4 * y) == (2 * t_0, -t_0)
+    assert diop_solve(4 * x + 6 * y - 4) == (3 * t_0 - 2, -2 * t_0 + 2)
+    assert diop_solve(4 * x + 6 * y - 3) == (None, None)
+    assert diop_solve(0 * x + 3 * y - 4 * z + 5) == (4 * t_0 + 5, 3 * t_0 + 5)
+    assert diop_solve(4 * x + 3 * y - 4 * z + 5) == (
+        t_0,
+        8 * t_0 + 4 * t_1 + 5,
+        7 * t_0 + 3 * t_1 + 5,
+    )
+    assert diop_solve(4 * x + 3 * y - 4 * z + 5, None) == (0, 5, 5)
+    assert diop_solve(4 * x + 2 * y + 8 * z - 5) == (None, None, None)
+    assert diop_solve(5 * x + 7 * y - 2 * z - 6) == (
+        t_0,
+        -3 * t_0 + 2 * t_1 + 6,
+        -8 * t_0 + 7 * t_1 + 18,
+    )
+    assert diop_solve(3 * x - 6 * y + 12 * z - 9) == (2 * t_0 + 3, t_0 + 2 * t_1, t_1)
+    assert diop_solve(6 * w + 9 * x + 20 * y - z) == (
+        t_0,
+        t_1,
+        t_1 + t_2,
+        6 * t_0 + 29 * t_1 + 20 * t_2,
+    )
```

Comment:
@oscarbenjamin this is probably what you meant in https://github.com/sympy/sympy/issues/17287#issuecomment-962292806. See also https://github.com/sympy/sympy/issues/17287#issuecomment-952527707

Reply:
Is it possible to configure black so that it does not force spaces around all operators?

Not being able to visually distinguish operator precedence is not good. If this is forced by black then that's a deal-breaker as far as I'm concerned.

Reply:
It seems not:
https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html
Black is intentionally hardly configurable. (I personally prefer that, but can see that is problematic for math-oriented code.) A possible way out is to put `fmt:on/off` around the code block.

Yapf is more configurable and seems to be able to handle this operator style:
https://github.com/google/yapf#knobs
Afaik yapfs is less popular, but was mentioned in the original issue https://github.com/sympy/sympy/issues/17287#issue-473721857

Reply:
Then I don't think we can use black. I commented here: https://github.com/psf/black/issues/538#issuecomment-962448644

Reply:
A post-process that closes up space on desired operators wouldn't be too hard, would it?

Reply:
> Then I don't think we can use black. I commented here: [psf/black#538 (comment)](https://github.com/psf/black/issues/538#issuecomment-962448644)

Thanks for investigating this and reporting it upstream!

> A post-process that closes up space on desired operators wouldn't be too hard, would it?

I wouldn't go for a post-process. The motivation to use black is to have standardised formatting, so if there is some post-formatting, you remove the reason of using black.